### PR TITLE
Added compatibility check for GPUDirect and GKE version

### PR DESCRIPTION
**Pull request by sharabiani**
### Changes
A condition added to check if GKE Version is compatible with GPUDirect for A3 High and A3 Mega machine types.
The changed are applied based on requirements in this document:  https://cloud.google.com/kubernetes-engine/docs/how-to/gpu-bandwidth-gpudirect-tcpx#requirements

### CODE REVIEW

File: modules/scheduler/pre-existing-gke-cluster/outputs.tf [hunk 1]
```diff
@@ -26,3 +26,8 @@ output "gke_cluster_exists" {
     data.google_container_cluster.existing_gke_cluster
   ]
 }
+
+output "gke_master_version" {
```
Comment (ankitkinra) [inline modules/scheduler/pre-existing-gke-cluster/outputs.tf:30]: `gke_version`

File: modules/compute/gke-node-pool/README.md
Comment (ankitkinra) [inline modules/compute/gke-node-pool/README.md:1]: Overall LGTM, just a few language nits.

File: modules/compute/gke-node-pool/gpu_direct.tf [hunk 2]
```diff
@@ -43,10 +49,25 @@ locals {
...
+
+  gke_version_regex = "(\\d+\\.\\d+)\\.(\\d+)-gke\\.(\\d+)" # GKE version format: 1.X.Y-gke.Z , regex output: ["1.X" , "Y", "Z"]
+
+  gke_version_parts = regex(local.gke_version_regex, var.gke_master_version)
+  gke_version_major = local.gke_version_parts[0]
+
+  min_gke_versions         = try(local.gpu_direct_setting[var.machine_type].min_gke_versions, null)
+  min_version              = try(contains(keys(local.min_gke_versions), local.gke_version_major), false) ? local.min_gke_versions[local.gke_version_major] : "1.0.0-gke.0"
+  min_version_parts        = regex(local.gke_version_regex, local.min_version)
+  gke_gpudirect_compatible = local.gke_version_parts[1] > local.min_version_parts[1] || (local.gke_version_parts[1] == local.min_version_parts[1] && local.gke_version_parts[2] >= local.min_version_parts[2])
```
Comment (ankitkinra) [inline modules/compute/gke-node-pool/gpu_direct.tf:70]: nit: can you please add a comment above this on with an example version ?

Comment (sharabiani) [reply modules/compute/gke-node-pool/gpu_direct.tf]: Already added an example in line 62

**Pull request merged**